src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Completion/RazorCompletionEndpoint.cs

```diff
@@ -22,19 +22,21 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Completion
     internal class RazorCompletionEndpoint : ICompletionHandler, ICompletionResolveHandler
     {
         private static readonly Container<string> DirectiveAttributeCommitCharacters = new Container<string>(" ");
-        private CompletionCapability _capability;
-        private readonly ILogger _logger;
-        private readonly ForegroundDispatcher _foregroundDispatcher;
-        private readonly DocumentResolver _documentResolver;
-        private readonly RazorCompletionFactsService _completionFactsService;
-        private readonly TagHelperCompletionService _tagHelperCompletionService;
-        private readonly TagHelperDescriptionFactory _tagHelperDescriptionFactory;
+
         private static readonly Command RetriggerCompletionCommand = new Command()
         {
             Name = "editor.action.triggerSuggest",
```

Comment:
I didn't see any issues, but maybe I just didn't trigger the correct case? If I understand you correctly this would try to trigger when I do (where ^ is trigger completion for NavMenu) `<N^>`? If that's the scenario it doesn't have any problems, but I also doesn't add the `</NavLink>` the way the existing tooling does.

Reply:
Ah, actually I recall how we used this. I believe the scenario it was used was when you auto-completed `a` => `asp-for="|"` in order to re-trigger completion at the `|`. That being said, now that VS doesn't support snippets we can't possibly re-trigger completion because the scenario we were used to doing it for doesn't apply 😄 . Ignore my comment here.